tmp_script.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+# Rung by passing in the function name to call as first parameter to the script.
+# E.g.  ./tmp_script.sh build_rust
+#
+# Be aware that this script will in a follow up PR be replaced with an ob runner.
+
+set -eux
+cd "$(dirname $0)"
+
+function install_async_bindgen_cli() {
+    cargo install \
+        --git https://github.com/xaynetwork/xayn_async_bindgen.git \
+        async-bindgen-gen-dart \
+        "$@"
+}
+
+function gen_dart_genesis_extension() {
+    cd discovery_engine_core
+    # Doesn't respect cargo config option, but good enough for us for now
+    "${CARGO_INSTALL_ROOT:-${CARGO_HOME:-$HOME/.cargo}}/bin/async-bindgen-gen-dart" \
+        --ffi-class XaynDiscoveryEngineBindingsFfi \
+        --genesis ../discovery_engine/lib/src/ffi/genesis.ffigen.dart
+    cd ..
+}
+
+function get_dart_deps() {
+    if [ "${_DART_PUB_GET_DONE:-0}" = 0 ]; then
+        _DART_PUB_GET_DONE=1
+        cd discovery_engine
+        dart pub get
+        cd ..
+    fi
+}
+
+function gen_dart_genesis() {
+    get_dart_deps
+    cd discovery_engine
+    dart run ffigen --config ffigen.yaml
+    cd ..
+}
+
+function gen_dart_build_runner() {
+    get_dart_deps
+    cd discovery_engine
+    dart run build_runner build --delete-conflicting-outputs
+    cd ..
+}
+
+function copy_ios_headers() {
+    # ios needs the original headers, as wee need to statically link in the
+    # binary code generated by rust (instead of loading it as a shared library)
+    cp ./discovery_engine_core/bindings/include/*.h ./discovery_engine_flutter/ios/Classes/
+}
+
+function codegen_workaround() {
+    if [ "${_CODEGEN_WORKAROUND_DONE:-0}" = 0 ]; then
+        _CODEGEN_WORKAROUND_DONE=1
+        cd discovery_engine_core
+        cargo check --quiet 2>/dev/null || :
+        cd ..
+    fi
+}
+
+function check_fmt_rust() {
+    cd discovery_engine_core
+    cargo +nightly fmt --all -- --check
```

Comment:
Would be more useful we can just run them to have the code as we want.

Reply:
I'm not sure what you want to say with that.

Reply:
If you mean we should run cargo fmt, I would disagree. It's a check command it should not modify your code, we use a nightly i.e. unstable version of rustfmt, it might brake your code with formatting.

But that's a discussion for the follow up PR not this PR.

Reply:
I mean that locally we can just run it and have the files in the correct format.